File: C++/Binary_search.cpp
#include<iostream>
using namespace std;
int main()
{
    int A[10] , n = 5 , l = 0 , h = n - 1 , mid , key;
    cout<<"Enter elements:";
    for(int i = 0 ; i < n ; i ++)
    {
        cin>>A[i];
    }
    cout<<"Enter key element:";
    cin>>key;
    while (l<=h)
    {
        mid = (l + h )/2;
        if(key == A[mid])
        {
            cout<<"Found at "<<mid;
            return 0;
        }
        else if(key<A[mid])
        h = mid - 1;
        else
        l = mid + 1;
    }

     cout<<"Not found";
    
    return 0;
}

File: Python/binarySearch.py
def binary_search(arr, target):
    left, right = 0, len(arr) - 1

    while left <= right:
        mid = left + (right - left) // 2  # Calculate the mid index

        # Check if the target is at the mid index
        if arr[mid] == target:
            return mid  # Return the index of the found item
        # If the target is greater, ignore the left half
        elif arr[mid] < target:
            left = mid + 1
        # If the target is smaller, ignore the right half
        else:
            right = mid - 1

    return -1  # Return -1 if the item is not found

# Example usage
arr = [10, 20, 30, 40, 50]  # Sample sorted array
target = 30  # Value to search for

result = binary_search(arr, target)

if result != -1:
    print(f"Element {target} found at index {result}.")
else:
    print(f"Element {target} not found in the array.")
